src/cached_digest.ml

```diff
@@ -27,14 +29,49 @@ let cache = lazy (
   | None ->
     { checked_key = 0
     ; table = Path.Table.create 1024
+    ; max_timestamp = 0.
     }
   | Some cache ->
     cache.checked_key <- cache.checked_key + 1;
     cache)
 
+let get_current_filesystem_time () =
+  let special_path = Path.relative Path.build_dir ".filesystem-clock" in
+  let modify_file () =
+    let contents =
+      match Io.read_file special_path with
+      | exception Sys_error _ -> None
+      | exception Unix.Unix_error (ENOENT, _, _) -> None
+      | v -> match Int.of_string (String.trim v) with
```

Comment:
This trim will not be necessary if you just use `Io.write` I think

Reply:
Sure. I added a newline to the end deliberately, but I agree there's no good reason to do it in this case.